zim/notebook/notebook.py

```diff
@@ -835,7 +835,7 @@ def _update_link_tag(self, elt, source, target, oldhref):
 
 		if elt.gettext() == elt.get('href'):
 			elt[:] = [link]
-		elif elt.gettext() == oldhref.parts()[-1] and len(elt) == 1:
+		elif oldhref.parts() and elt.gettext() == oldhref.parts()[-1] and len(elt) == 1:
```

Comment:
I think you meant to type `newhref` instead of `oldhref` here - that fixes the issue.

Reply:
I added a check for `newhref` being empty too (the case you ran into). While testing yesterday I had an issue related to `oldhref` which was obviously a different problem.
This would also mean I was not able to reproduce your initial problem. Anyway, the parts list maybe empty and we have to check for this case.